yarn-project/pxe/src/simulator_oracle/index.ts

```diff
@@ -303,70 +310,141 @@ export class SimulatorOracle implements DBOracle {
+      );
+
+      let { currentTagggingSecrets } = searchState;
+      const { maxIndexesToCheck, secretsToIncrement, initialSecretIndexes } = searchState;
+
+      while (currentTagggingSecrets.length > 0) {
+        // 2. Compute tags using the secrets, recipient and index. Obtain logs for each tag (#9380)
+        const currentTags = currentTagggingSecrets.map(taggingSecret => taggingSecret.computeTag(recipient));
+        const logsByTags = await this.aztecNode.getLogsByTags(currentTags);
+        const newTaggingSecrets: IndexedTaggingSecret[] = [];
+        logsByTags.forEach((logsByTag, logIndex) => {
+          const { secret: currentSecret, index: currentIndex } = currentTagggingSecrets[logIndex];
+          const currentSecretAsStr = currentSecret.toString();
+          this.log.debug(
+            `Syncing logs for recipient ${recipient}, secret ${currentSecretAsStr}:${currentIndex} at contract: ${contractName}(${contractAddress})`,
           );
-        }
-      });
-      // 4. Consolidate in db and replace initial appTaggingSecrets with the new ones (updated indexes)
-      await this.db.incrementTaggingSecretsIndexes(
-        newTaggingSecrets.map(secret => new TaggingSecret(secret.secret, recipient)),
+          // 3.1. Append logs to the list and increment the index for the tags that have logs (#9380)
+          if (logsByTag.length > 0) {
+            this.log.verbose(
+              `Found ${
+                logsByTag.length
+              } logs for secret ${currentSecretAsStr} as recipient ${recipient}. Incrementing index to ${
+                currentIndex + 1
+              } at contract: ${contractName}(${contractAddress})`,
+            );
+            logs.push(...logsByTag);
+
+            if (currentIndex >= initialSecretIndexes[currentSecretAsStr]) {
+              // 3.2. Increment the index for the tags that have logs, provided they're higher than the one
+              // we have stored in the db (#9380)
+              secretsToIncrement[currentSecretAsStr] = currentIndex + 1;
+              // 3.3. Slide the window forwards if we have found logs beyond the initial index
+              maxIndexesToCheck[currentSecretAsStr] = currentIndex + INDEX_OFFSET;
+            }
+          }
+          // 3.4 Keep increasing the index (inside the window) temporarily for the tags that have no logs
+          // There's a chance the sender missed some and we want to catch up
+          if (currentIndex < maxIndexesToCheck[currentSecretAsStr]) {
+            const newTaggingSecret = new IndexedTaggingSecret(currentSecret, currentIndex + 1);
+            newTaggingSecrets.push(newTaggingSecret);
+          }
+        });
+        await this.db.setTaggingSecretsIndexesAsRecipient(
+          Object.keys(secretsToIncrement).map(
+            secret => new IndexedTaggingSecret(Fr.fromString(secret), secretsToIncrement[secret]),
+          ),
+        );
+        currentTagggingSecrets = newTaggingSecrets;
```

Comment:
I'll open a new issue about this: we should only increase the index if we actually find a note in the logs that we processed for that index. We could otherwise get spammed (assuming someone knows the tagging secret).

Reply:
Created #9953